### ISSUE TITLE
[ENH] should CI for a `pyproject.toml` change turn off incremental testing?

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#5544]
Question for CI in the case where we update `pyproject.toml`, such as this one:
https://github.com/sktime/sktime/pull/5538

In such a case, should we not run all the tests while turning *off* incremental testing?

That is, to ensure that estimators depending on the extra do not break, or detect that they break.

This could be achieved in two locations - GHA, or `run_tests_for_class`, adding one more conditional so it always defaults to `True` if the `pyproject.toml` has changed.

FYI @yarnabrina, @benHeid

**Comment by yarnabrina** [comment]
Makes sense (and I'll agree if you disable even sub-sampling). But it'll possibly lead to long duration in my opinion unless https://github.com/sktime/sktime/issues/5477 is solved.

**Comment by fkiraly** [comment]
One could argue, the deps are so important so we should test really everything - they are after all relatively rare.
